core/store/migrations/migration1586949323/migrate.go

```diff
@@ -0,0 +1,32 @@
+package migration1586949323
+
+import (
+	"github.com/jinzhu/gorm"
+)
+
+// Migrate makes txes.hash a unique index
+// Due to faulty logic, there are some duplicates out in the wild (even though this should never happen).
+// In the case of duplicate hashes, we keep the earliest row for each hash (by ID) and delete all other duplicate rows.
```

Comment:
In the case of a Tx duplicate, do we need to point TxAttempts associated with the deleted Txs at the remaining Tx record?

Reply:
I don't know. It depends I suppose on how they came into being.

Reply:
@se3000 I thought about this some more, and IMO the safest thing to do is to delete the transaction attempts rather than trying to copy them over.

My reasoning for this is that those transaction attempts would never have been accessible anyway because trying to look them up through the txes.hash would have always returned only the first tx.

Reply:
Otherwise we risk trying to reattempt transactions long since committed.